### PR TITLE
[FLINK-11140][streaming] Fix empty child path check in Buckets

### PR DESCRIPTION
**Pull request by Matrix42**
## What is the purpose of the change

Fix empty child path check in Buckets

## Brief change log

add check for empty child path in Buckets

## Verifying this change

BucketsTest#testAssembleBucketPath()

## Does this pull request potentially affect one of the following parts:

- Dependencies (does it add or upgrade a dependency): ( no)
- The public API, i.e., is any changed class annotated with @Public(Evolving): (no)
- The serializers: ( no)
- The runtime per-record code paths (performance sensitive): (no )
- Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no )
- The S3 file system connector: (no )

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/Buckets.java [hunk 1]
```diff
@@ -304,6 +304,10 @@ void close() {
 	}
 
 	private Path assembleBucketPath(BucketID bucketId) {
+		String child = bucketId.toString();
+		if ("".equals(child)) {
+			return basePath;
+		}
 		return new Path(basePath, bucketId.toString());
```
Comment (yanghua) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/Buckets.java]: Now that you have given `bucketId.toString()` to the variable `child`, you can change it to: `return new Path(basePath, child);`

**Comment by zentol** [comment]
IMO an empty bucket Id doesn't make any sense as it effectively disables bucketing.
As far as i can tell this is an implementation error on the user-side requiring no change on our side.
@kl0u WDYT?

**Comment by zentol** [comment]
Oh for heaven's sake this is one of our classes isn't it. god damn it.

**Comment by aljoscha** [comment]
@zentol I think this is a valid fix because it's our class, right?

@Matrix42 Can you please reopen the PR?

**Comment by kl0u** [comment]
@Matrix42 I think this is a valid issue. I will have a look to the PR now and thanks for reporting it.

**Comment by Matrix42** [comment]
@zentol @kl0u thanks for your review, I will reopen this PR.

**Comment by kl0u** [comment]
Sorry, I just saw that you addressed the comments. I will review soon.

**Pull request closed without merging**